array.c

```diff
@@ -1996,6 +1997,23 @@ rb_ary_last(int argc, const VALUE *argv, VALUE ary) // used by parse.y
     }
 }
 
+static VALUE
+ary_fetch(VALUE ary, long idx, VALUE ifnone, bool block_given)
```

Comment:
Why is this function located away from its only caller, `append_fetch_values_single`?
Maybe you forgot to update `rb_ary_fetch` to use it too?